### ISSUE TITLE
Section "collapsible" element is not accessible with keyboard navigation

### ISSUE THREAD

**Issue by evansjohnson** [palantir/blueprint#6872]
#### Steps to reproduce
https://blueprintjs.com/docs/#core/components/section
- Toggle on "Collapsible"
- Attempt to access collapsible header with keyboard

#### Actual behavior
Cannot access collapsible header with keyboard

#### Expected behavior
Should be able to access all interactive elements with keyboard

**Comment by mm-wang** [comment]
Do you mean the section title overall? Collapsible seems to work with tabbing through.

https://github.com/user-attachments/assets/64f47ca1-bf94-44d0-bfe9-5685ef59bd61
